core/src/trezor/pin.py

```diff
@@ -1,30 +1,27 @@
-from typing import Any
+from typing import TYPE_CHECKING
 
-from trezor import utils
+if TYPE_CHECKING:
+    from typing import Any
+    from trezor.ui.layouts.common import ProgressLayout
 
-_previous_progress: int | None = None
 _previous_seconds: int | None = None
+_previous_remaining: str | None = None
+_progress_layout: ProgressLayout | None = None
```

Comment:
ISTM we can now be super conservative in terms of imports.

it should be possible to:
(a) have a local copy of `backlight_fade()`
(b) import `trezorui` locally, instantiate a global `display = Display()` (it's a single block allocation that will remain static forever)
(c) import `trezorui2` locally and create a global `progress = trezorui2.show_progress()`
(d) run it via a bunch of plain functions

OTOH we are _almost_ at the point where we could skip the middleman and do this whole progress callback directly from Rust, so maybe instead I'll create a separate issue to do that